src/Analyzers/CSharp/Analyzers/NewLines/ConditionalExpressionPlacement/ConditionalExpressionPlacementDiagnosticAnalyzer.cs

```diff
@@ -11,108 +11,85 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 
-namespace Microsoft.CodeAnalysis.CSharp.NewLines.ConditionalExpressionPlacement
+namespace Microsoft.CodeAnalysis.CSharp.NewLines.ConditionalExpressionPlacement;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class ConditionalExpressionPlacementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
 {
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    internal sealed class ConditionalExpressionPlacementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    public ConditionalExpressionPlacementDiagnosticAnalyzer()
+        : base(IDEDiagnosticIds.ConditionalExpressionPlacementDiagnosticId,
+               EnforceOnBuildValues.ConditionalExpressionPlacement,
+               CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression,
+               new LocalizableResourceString(
+                   nameof(CSharpAnalyzersResources.Blank_line_not_allowed_after_conditional_expression_token), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
     {
-        public ConditionalExpressionPlacementDiagnosticAnalyzer()
-            : base(IDEDiagnosticIds.ConditionalExpressionPlacementDiagnosticId,
-                   EnforceOnBuildValues.ConditionalExpressionPlacement,
-                   CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression,
-                   new LocalizableResourceString(
-                       nameof(CSharpAnalyzersResources.Blank_line_not_allowed_after_conditional_expression_token), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
-        {
-        }
+    }
 
-        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
-            => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
```

Comment:
>SyntaxTreeWithoutSemanticsAnalysis [](http://example.com/codeflow?start=42&length=34)

Would like to understand a bit better why the switch from tree based analyzer to semantic based analyzer is good. I thought the syntax based ones would come back sooner, allowing quicker lightbulb data. Or is that not the case?

Reply:
Talked offline. From Cyrus, it sounds like it's ok if this analyzer is delayed a bit, and that the tradeoff with that versus needing to walk/realize the whole tree is beneficial.